torchdynamo/symbolic_convert.py

```diff
@@ -1362,6 +1368,23 @@ def LOAD_CLOSURE(self, inst):
         assert inst.argval in self.cell_and_freevars()
         self.push(self.closure_cells[inst.argval])
 
+    def is_from_parent(self, var: VariableTracker):
+        for k, v in self.parent.symbolic_locals.items():
+            if var.mutable_local is v.mutable_local:
+                return True
+        return False
+
+    def replace_all(self, oldvar: VariableTracker, newvar: VariableTracker):
+        if isinstance(
+            oldvar, (ListVariable, ConstDictVariable)
+        ) and self.is_from_parent(oldvar):
```

Comment:
I think we may just want to do this all the time.  It could be the case that we are mutating something nested inside a data structure.

Reply:
Oh, yes! if this variable is not from parent, then parent translator's ```update_locals_and_stack``` do nothing.